onnx/shape_inference/implementation.cc

```diff
@@ -253,9 +254,79 @@ std::string GetModelLocalFunctionsMapIdentifier(const std::string& domain, const
   return domain + ":" + func_name;
 }
 
+// InferredTypes: abstracts the differences between FunctionProto and GraphProto
+// for inference. For GraphProto, inferred types are stored in the GraphProto
+// but FunctionProto does not have a place to store inferred types. So, we
+// use a temporary vector (for the duration of inference) to store these.
+class InferredTypes {
+ public:
+  explicit InferredTypes(GraphProto* graph = nullptr) : graph_ptr(graph) {}
```

Comment:
Do we need to initialize types?

Reply:
No, the default constructor should be fine.